### PR TITLE
feat(core): add the `offset` into `FindOneOptions`

### LINKED ISSUE #3292: Add an "offset" option to "em.findOne()"
**Is your feature request related to a problem? Please describe.**
I want to be able to find a single random child from a parent. In my app there are "Deck" and "Card" entities, and I want to be able to choose a random card from a deck.

**Describe the solution you'd like**
I'd like to be able to do something like so:
```typescript
let card = await em.findOne(
    Card,
    { deck: "my_deck", alreadyPlayed: false },
    { orderBy: { order: QueryOrder.ASC }, offset: randomNumber() }
);
```
However "offset" is not an available option.

Another solution could be if there was a such thing as "QueryOrder.Random".

**Describe alternatives you've considered**
I can work around this by using `em.find()` with a `limit` instead:
```typescript
let card = await em.find(
    Card,
    { deck: "my_deck", alreadyPlayed: false },
    { orderBy: { order: QueryOrder.ASC }, offset: randomNumber(), limit: 1 }
)?.[0];
```

However it would be more convenient if `findOne()` had an "offset" option.

### PR DESCRIPTION
**Pull request by Phosphorus-M**
Just add the possibility of using findOne with an offset in case that is needed.
This PR doesn't have breaking changes.

Closes #3292

### CODE REVIEW

File: tests/issues/GH3292.test.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+test('test findOne but with a offset', async () => {
+  mockLogger(orm, ['query', 'query-params']);
+
+  const author = new Author2('Bartleby', 'bartelby@writer.org');
+  new Book2('My Life on The Wall, part 1', author);
+  const book2 = new Book2('My Life on The Wall, part 2', author);
+
+  await orm.em.fork().persistAndFlush(author);
+
+  async function requestCommonService() {
```
Comment (B4nan) [inline tests/issues/GH3292.test.ts:45]: same here

File: tests/issues/GH3292.test.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      const titleA = await requestCommonService();
+      const titleB = b.title;
+      return { titleA, titleB };
+    });
+
+    expect(titleA).toEqual(titleB);
+});
+
+test('test findOne but with a offset', async () => {
+  mockLogger(orm, ['query', 'query-params']);
```
Comment (B4nan) [inline tests/issues/GH3292.test.ts:37]: same here

File: tests/issues/GH3292.test.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    async function requestCommonService() {
+      const b = await orm.em.findOne(Book2, {});
+      return b?.title;
+    }
+
+    const { titleA, titleB } = await orm.em.transactional(async () => {
+      const b = await orm.em.findOneOrFail(Book2, book.uuid);
+      const titleA = await requestCommonService();
+      const titleB = b.title;
+      return { titleA, titleB };
```
Comment (B4nan) [inline tests/issues/GH3292.test.ts:30]: this does not make much sense, you are not testing anything here, this will always pass

**Comment by Phosphorus-M** [comment]
Do you think that now it's better?

File: tests/issues/GH3292.test.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import { Author2, Book2 } from '../entities-sql';
+
+let orm: MikroORM;
+
+beforeAll(async () => orm = await initORMPostgreSql());
+beforeEach(async () => orm.schema.clearDatabase());
+afterAll(() => orm.close(true));
+
+test('test findOne without a offset', async () => {
+
```
Comment (B4nan) [inline tests/issues/GH3292.test.ts:12]:
```suggestion
test('test findOne without a offset', async () => {
```

File: tests/issues/GH3292.test.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  await orm.em.fork().persistAndFlush(author);
+
+  const myBook = await orm.em.transactional(async () => await orm.em.findOne(Book2, {}));
+
+  expect(myBook?.title).toEqual(book.title);
+});
+
+test('test findOne but with a offset', async () => {
+
```
Comment (B4nan) [inline tests/issues/GH3292.test.ts:25]:
```suggestion
test('test findOne but with a offset', async () => {
```

**Review by B4nan — APPROVED** [review]

**Review by B4nan — COMMENTED** [review]
code style

**Comment by Phosphorus-M** [comment]
Oh, it's true.

Sorry to bother but could you consider adding "hacktoberfest-accepted" as a label? 
If don't you consider it a enough don't worry I can understand it

**Comment by B4nan** [comment]
Oh its hacktoberfest already, didnt even realize. Yeah I dont mind, however small the contribution was, it's still valueable :]

**Pull request merged**